code/game/objects/items/robot/robot_parts.dm

```diff
@@ -201,38 +205,61 @@
 				user << "<span class='warning'>This [W] does not seem to fit.</span>"
 				return
 
-			var/mob/living/silicon/robot/O = new /mob/living/silicon/robot(get_turf(loc), TRUE)
-			if(!O)	return
+			if(!src.head.law_manager)
 
-			user.drop_item()
+				if(!is_alien_whitelisted(M.brainmob, "Machine") && config.usealienwhitelist)
+					user << "<span class='warning'>This [W] does not seem to fit.</span>"
+					return
 
-			O.mmi = W
-			O.invisibility = 0
-			O.custom_name = created_name
-			O.updatename("Default")
+				var/mob/living/carbon/human/unbranded_frame/new_shell = new(get_turf(loc), TRUE, "Unbranded Frame")
```

Comment:
Why are you setting species right after this line instead of just passing the species in the constructor?